tagstudio/src/qt/modals/ffmpeg_checker.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import math
+from pathlib import Path
+import subprocess
+
+from PIL import Image, ImageQt
+from pydub.utils import which
+from PySide6.QtCore import Signal, Qt, QUrl
+from PySide6.QtGui import QPixmap, QDesktopServices
+from PySide6.QtWidgets import QMessageBox
+
+
+class FfmpegChecker(QMessageBox):
+    """A warning dialog for if FFmpeg is missing."""
+
+    help_url = "https://github.com/TagStudioDev/TagStudio/blob/Alpha-v9.4/doc/utilities/ffmpeg.md"
+
+    def __init__(self):
+        super().__init__()
+
+        self.setIcon(QMessageBox.Warning)
+
+        self.setText("Warning: Could not find FFmpeg installation")
+        self.setStandardButtons(
+            QMessageBox.Help | QMessageBox.Ignore | QMessageBox.Cancel
+        )
+        self.setDefaultButton(QMessageBox.Ignore)
+        # Enables the cancel button but hides it to allow for click X to close dialog
+        self.button(QMessageBox.Cancel).hide()
+
+        # Blocks other application interactions until resolved
+        self.setWindowModality(Qt.ApplicationModal)
+        self.setWindowTitle("Warning: Missing dependency")
+
+        self.ffmpeg_missing = True
+        self.ffprobe_missing = True
+
+    def installed(self):
+        """Checks if both FFmpeg and FFprobe are installed and in the PATH."""
+        # Same checker that ffmpeg-python uses
+        if which("ffmpeg"):
```

Comment:
I dont like this implementation of which, it could just have been a method in your class, but since it ships with ffmpegs install it is fine to use. But, it uses `os.name` for platform checking, and I have seen many discussion about this usage and how it is better to use `platform.system()` to get the current platform name. 

It also uses os.path instead of `pathlib` 

I would like @CyanVoxel to comment on this. I can write the method if that is wanted as a new PR after this is merged. there is currently nothing wrong about this usage.

Reply:
Could `shutil.which()` be used here instead?

Reply:
I did not know the Python stdlib had a built in which command, using that instead.